java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeSECheck.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.bytecode.se;
+
+import org.objectweb.asm.Opcodes;
+import org.sonar.java.bytecode.cfg.BytecodeCFGBuilder;
+import org.sonar.java.se.ProgramState;
+import org.sonar.java.se.constraint.ObjectConstraint;
+import org.sonar.java.se.symbolicvalues.SymbolicValue;
+
+public interface BytecodeSECheck {
+
+  default ProgramState checkPreStatement(CheckerDispatcher dispatcher, BytecodeCFGBuilder.Instruction inst) {
+    return dispatcher.getState();
+  }
+
+  default ProgramState checkPostStatement(CheckerDispatcher dispatcher, BytecodeCFGBuilder.Instruction inst) {
+    return dispatcher.getState();
+  }
+
+  class NullnessCheck implements BytecodeSECheck {
+
+    @Override
+    public ProgramState checkPreStatement(CheckerDispatcher dispatcher, BytecodeCFGBuilder.Instruction inst) {
+      ProgramState state = dispatcher.getState();
+      if (isInvokeOnObjectRef(inst)) {
+        SymbolicValue objectRef = state.peekValue(/*TODO peek the n+1 value with n arity of the method invoked*/);
+        ObjectConstraint constraint = state.getConstraint(objectRef, ObjectConstraint.class);
+        if (constraint != null && constraint.isNull()) {
+//       Exceptional Yield should be added : context.addExceptionalYield(currentVal, programState, JAVA_LANG_NPE, this);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)